src/smartpeak/source/core/SequenceProcessor.cpp

```diff
@@ -217,7 +217,7 @@ namespace SmartPeak
     try {
       std::list<std::future<void>> futures;
       LOGD << "Spawning workers...";
-      for (size_t i = 0; i < n_workers; ++i) {
```

Comment:
I think this should actually be `n_workers` here as we want the user to be able to specify the number of threads.  If the `n_threads` param is left unspecified in the parameters.csv file, `getNumWorkers` should set this to the maximum number of threads that is available on the system -1 for the GUI's dedicated thread.

Reply:
I have reverted the changes, the changes here doesn't seem to be thread safe and require some sort of locking before resources are freed. Under certain circumstances it's causing an undefined behaviour and a possible crash. I will be investigating this separately.

Reply:
Should there be a check that the number of workers is less than the number of requested injections in the method that determines the number of wokers?